src/main/java/org/apache/maven/plugins/dependency/fromDependencies/CopyDependenciesMojo.java

```diff
@@ -266,6 +274,15 @@ protected void copyArtifact(
         }
         try {
             copyUtil.copyArtifactFile(artifact, destFile);
+
+            // Copy the signature file if the parameter is enabled
+            if (copySignatures) {
+                File signatureFile = new File(artifact.getFile().getAbsolutePath() + ".asc");
+                if (signatureFile.exists()) {
+                    File signatureDestFile = new File(destDir, destFileName + ".asc");
+                    copyUtil.copyFile(signatureFile, signatureDestFile);
+                }
+            }
         } catch (IOException e) {
```

Comment:
Maybe we need to introduce a new catch IOException for the signature files. 

The case that I have in mind: Something goes wrong during copying the signature, but this catch introduces the message that something goes wrong during copying the artifact and hides the root cause.

Reply:
Yes, this is a MojoSomethingException (either Failure or Execution). Don't have javadocs in front of me, but it should be thrown.